seqr/management/commands/lift_project_to_hg38_tests.py

```diff
@@ -0,0 +1,269 @@
+
+        with self.assertRaises(CommandError) as ce:
+            call_command('lift_project_to_hg38', u'--project={}'.format(PROJECT_NAME),
+                    '--es-index={}'.format(ELASTICSEARCH_INDEX))
+
+        self.assertEqual(ce.exception.message, 'Matches not found for ES sample ids: ID_NOT_EXIST.')
+
+        calls = [
+            mock.call(u'Updating project genome version for {}'.format(PROJECT_NAME)),
+            mock.call('Validating es index test_index')
+        ]
+        mock_logger.info.assert_has_calls(calls)
+
+        mock_get_es_samples.assert_called_with(ELASTICSEARCH_INDEX)
+
+    @mock.patch('seqr.management.commands.lift_project_to_hg38.logger')
+    @mock.patch('seqr.management.commands.lift_project_to_hg38.get_elasticsearch_index_samples')
+    def test_command_error_missing_indvididuals(self, mock_get_es_samples, mock_logger):
+        mock_get_es_samples.return_value = SAMPLE_IDS, INDEX_METADATA
+
+        # Create a new individual belonging to family 1 but has no sample
+        family = Family.objects.get(pk=1)
+        individual = Individual.objects.create(family = family, individual_id = "NO_MATCHED_SAMPLE_IND_ID")
+        individual.save()
+        sample = Sample.objects.create(
+            individual = individual,
+            sample_type = "WES",
+            dataset_type = "VARIANTS",
+            sample_id = "NO_MATCHED_SAMPLE_ID",
+            is_active = True,
+            loaded_date = timezone.now()
+        )
+        sample.save()
+
+        with self.assertRaises(CommandError) as ce:
+            call_command('lift_project_to_hg38', u'--project={}'.format(PROJECT_NAME),
+                    '--es-index={}'.format(ELASTICSEARCH_INDEX))
+
+        self.assertEqual(ce.exception.message, 'The following families are included in the callset but are missing some family members: 1 (NO_MATCHED_SAMPLE_IND_ID).')
+
+        calls = [
+            mock.call(u'Updating project genome version for {}'.format(PROJECT_NAME)),
+            mock.call('Validating es index test_index'),
+        ]
+        mock_logger.info.assert_has_calls(calls)
+
+        mock_get_es_samples.assert_called_with(ELASTICSEARCH_INDEX)
+
+    @mock.patch('seqr.management.commands.lift_project_to_hg38.logger')
+    @mock.patch('seqr.management.commands.lift_project_to_hg38.get_elasticsearch_index_samples')
+    def test_command_error_missing_families(self, mock_get_es_samples, mock_logger):
+        mock_get_es_samples.return_value = SAMPLE_IDS, INDEX_METADATA
+
+        # Create a new SavedVariant record which belongs to a family missed
+        family = Family.objects.get(pk=3)
+        saved_variant = SavedVariant.objects.create(
+            xpos_start = 21005543353,
+            xpos_end = 21005543356,
+            ref = "GAGA",
+            alt = "G",
+            variant_id = "21-5543353-GAGA-G",
+            family = family
+        )
+        saved_variant.save()
```

Comment:
Never create test data via the django API, except under extenuating circumstances. Instead, change the sample IDs that are returned from the mock

Reply:
Removed creating test data. Added a mocked SavedVariant.objects.filter() return_value.